src/components/HeaderLink.js

```diff
@@ -1,34 +1,50 @@
 import React from 'react';
 import PropTypes from 'prop-types';
-import { NavLink } from 'react-router-dom';
 import classnames from 'classnames';
 
-export const HeaderLink = ({ children, className, exact, href, ...props }) => {
-  const classes = classnames(['header-link', className]);
+export const HeaderLink = ({ children, className, href, link, ...props }) => {
+  const attributes = {
+    className: classnames(['header-link', className]),
+    ...props,
+  };
 
   if (!href) {
-    return <span className={classes} {...props}>{children}</span>;
+    return <span {...attributes}>{children}</span>;
   }
 
-  if (href.startsWith('http')) {
-    return <a className={classes} href={href} {...props}>{children}</a>;
+  if (link && !href.startsWith('http')) {
+    const {
+      component: Link,
+      linkProps: { hrefAttribute = 'href', ...linkProps },
+    } = link;
+
+    const attributesLink = {
+      ...attributes,
+      [hrefAttribute]: href,
+      ...linkProps,
+    };
+
+    return <Link {...attributesLink}>{children}</Link>;
```

Comment:
I do not know if it is better or not. I was just giving it a try:

```suggestion
  if (link && !href.startsWith('http')) {
    const {
      component: Component,
      linkProps: { hrefAttribute = 'href', ...linkProps },
    } = link;

    return (
      <Component
        {...attributes}
        {...{ [hrefAttribute]: href }}
        {...linkProps}
      >
        {children}
      </Component>
    );
```